### ISSUE TITLE
Send Emails

### ISSUE THREAD

**Issue by biqas** [ory/kratos#1168]
**Is your feature request related to a problem? Please describe.**
The problem is that there are some other template engines which are more suitable to compose html emails, also like here described #1030 the need to hand over to some other applications which handle those things from another view angel.

**Describe the solution you'd like**

I would like to have a possibility to either push based or pull based mechanism available in kratos.

*push:*
kratos pushes data to an endpoint during email sending if so configured. the endpoint must fulfill to accept a post request with a defined structure + data (identity and related data, which is configurable). the endpoint must also respond with a status: send or an error message with json payload. or if configured not to wait for a respond at all or ignore the respond.

*pull:*
kratos provide an API where you can pull a list which identities or actions need an email delivery and for which kind of process/workflow it is needed. The processor can decide on his own when to process and how to process and which data needs to be considered because he can make additional API calls to get the rest of the data which is needed. Additionally, an API endpoint is required so the processor can inform back kratos if an email was successfully delivered or an error accrued, but this can be optional.

the *pull* based would be preferred.

**Comment by zepatrik** [comment]
Just to make it clear, I do not want to implement a fully fledged pub/sub system within Kratos.
Is there maybe some standard that allows Kratos to post to a generic pub/sub system if desired, but also to go without one if the setup is able to handle all messages without that? I guess that would be the easiest and most flexible solution then.
Alternatively, you can use the solution described in #1030 to write a middleware server that fits your push/pull/... requirements.
So basically, if there is a standard that fits our use case we can use it, otherwise invent one.

**Comment by biqas** [comment]
@zepatrik I'm not proposing the implement a pub/sub system. 
Mybe let me simplify, what I'm requesting is that kratos should provide an endpoint where you can pick up records which need email processing. Nothing more.

**Comment by aeneasr** [comment]
I think a push / RPC model would be much more preferable as we can implement proper retry mechanisms, you reduce load on the database, and it's near-time and not batch based. What do you think?

**Comment by biqas** [comment]
@aeneasr yes of course looking from the user experience would be much better push based, but it brings some problems with it :), like what will you do if it fails also with retries, but I guess the same question could be raised also today with cratos right?
But having a push /RPC would be fine, do you have already in mind which data/payload you would provide in the call?

**Comment by aeneasr** [comment]
I'd probably just wire the data models we have today already, so stuff in courier/template
